spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1751,15 +1751,15 @@ private void sleepFor(Duration duration) {
 
 		private void enforceRebalanceIfNecessary() {
 			try {
-				if (KafkaMessageListenerContainer.this.enforceRebalanceRequested.get()) {
+				if (KafkaMessageListenerContainer.this.thisOrParentContainer.enforceRebalanceRequested.get()) {
 					final String enforcedRebalanceReason = String.format("Enforced rebalance requested for container: %s",
 							KafkaMessageListenerContainer.this.getListenerId());
-					KafkaMessageListenerContainer.this.logger.info(enforcedRebalanceReason);
+					ListenerConsumer.this.logger.info(enforcedRebalanceReason);
```

Comment:
I don't think we need `ListenerConsumer` prefix when we deal in the context of exactly `ListenerConsumer`.
I also think that those local `final` for variables give us a little more noise reading this code.